File: leetcode/226.invert-binary-tree/go/inverttree.go
package inverttree

type TreeNode struct {
	Val   int
	Left  *TreeNode
	Right *TreeNode
}

func invertTree(root *TreeNode) *TreeNode {
	swapNodes(root)
	return root
}

func swapNodes(root *TreeNode) {
	if root == nil {
		return
	}

	temp := root.Left
	root.Left = root.Right
	root.Right = temp

	swapNodes(root.Left)
	swapNodes(root.Right)
}

File: leetcode/226.invert-binary-tree/ts/inverttree.ts
class TreeNode {
    val: number
    left: TreeNode | null
    right: TreeNode | null
    constructor(val?: number, left?: TreeNode | null, right?: TreeNode | null) {
        this.val = (val === undefined ? 0 : val)
        this.left = (left === undefined ? null : left)
        this.right = (right === undefined ? null : right)
    }
}

const invertTree = (root: TreeNode | null): TreeNode | null => {
    swapNodes(root)
    return root
}

const swapNodes = (root: TreeNode | null) => {
    if (root === null) return

    const temp = root.left
    root.left = root.right
    root.right = temp

    swapNodes(root.left)
    swapNodes(root.right)
}

export default invertTree
